pkg/cmd/config/delete-garden.go

```diff
@@ -0,0 +1,96 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+
+	"github.com/spf13/cobra"
+)
+
+// NewCmdConfigDeleteGarden returns a new (config) delete-garden command.
+func NewCmdConfigDeleteGarden(f util.Factory, ioStreams util.IOStreams) *cobra.Command {
+	o := NewDeleteGardenOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:   "delete-garden",
+		Short: "delete Garden from gardenctl configuration",
+		Long:  "Delete Garden from gardenctl configuration. E.g. \"gardenctl config delete-garden my-garden\" to delete my-garden",
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			suggestions, err := validGardenArgsFunction(f, args)
+			if err != nil {
+				fmt.Fprintln(o.IOStreams.ErrOut, err.Error())
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+
+			return util.FilterStringsByPrefix(toComplete, suggestions), cobra.ShellCompDirectiveNoFileComp
+		},
+		RunE: base.WrapRunE(o, f),
+	}
+
+	return cmd
+}
+
+// Run executes the command
+func (o *DeleteGardenOptions) Run(f util.Factory) error {
```

Comment:
For the reason I have explained above I would like to remove all input parameters from the Run method of the Options intrface. We should store everthing we need for running the command in the Complete method on the options object and validate the the require data are the in the Validation method. In this case we would store the configuration object which we need here for the command execution.